src/Microsoft.Data/DataFrame.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// A DataFrame 
+    /// </summary>
+    public partial class DataFrame
+    {
+        internal DataFrameTable _table;
+        public DataFrame()
+        {
+            _table = new DataFrameTable();
+        }
+
+        public long RowCount => _table.RowCount;
+        public int ColumnCount => _table.ColumnCount;
+        public IList<string> Columns()
+        {
+            var ret = new List<string>();
+            for (int ii = 0; ii < ColumnCount; ii++)
+            {
+                ret.Add(_table.Column(ii).Name);
+            }
+            return ret;
+        }
+        public BaseDataFrameColumn Column(int index) => _table.Column(index);
```

Comment:
blank lines between each please and blank line after `}` unless followed by `}` is our pattern I think